SOPT-iOS/Projects/Domain/Sources/UseCase/DailySoptuneUseCase.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  DailySoptuneUseCase.swift
+//  Domain
+//
+//  Created by Jae Hyun Lee on 9/23/24.
+//  Copyright © 2024 SOPT-iOS. All rights reserved.
+//
+
+import Combine
+
+import Core
+
+public protocol DailySoptuneUseCase {
+    var todaysFortuneCard: PassthroughSubject<DailySoptuneCardModel, Never> { get }
+    var randomUser: PassthroughSubject<[PokeRandomUserInfoModel], Never> { get }
+    var pokedResponse: PassthroughSubject<PokeUserModel, Never> { get }
+    
+    func getTodaysFortuneCard()
+    func getRandomUser()
+    func poke(userId: Int, message: PokeMessageModel, isAnonymous: Bool)
+}
+
+public class DefaultDailySoptuneUseCase {
+    
+    public let repository: DailySoptuneRepositoyInterface
+    public let cancelBag = CancelBag()
+    
+    public let todaysFortuneCard = PassthroughSubject<DailySoptuneCardModel, Never>()
+    public let randomUser = PassthroughSubject<[PokeRandomUserInfoModel], Never>()
+    public let pokeMessages = PassthroughSubject<PokeMessagesModel, Never>()
+    public let pokedResponse = PassthroughSubject<PokeUserModel, Never>()
+    public let errorMessage = PassthroughSubject<String?, Never>()
+    
+    public init(repository: DailySoptuneRepositoyInterface) {
+        self.repository = repository
+    }
+}
+
+extension DefaultDailySoptuneUseCase: DailySoptuneUseCase {
+    
+    public func getTodaysFortuneCard() {
+        repository.getTodaysFortuneCard()
+            .sink { event in
+                print("GetTodaysFortuneCard State: \(event)")
+            } receiveValue: { [weak self] todaysFortuneCard in
+                self?.todaysFortuneCard.send(todaysFortuneCard)
+            }.store(in: cancelBag)
+    }
```

Comment:
Combine+에 withUnretained 메소드를 사용하면 좋을 것 같아요 ! 메모리 누수 방지는 같지만 가독성 측면에서 더 좋을 것 같습니다 ~ [weak self] 있는 다른 부분에서도 적용하면 좋을 듯 !_!

```suggestion
    public func getTodaysFortuneCard() {
        repository.getTodaysFortuneCard()
            .withUnretained(self)
            .sink { event in
                print("GetTodaysFortuneCard State: \(event)")
            } receiveValue: { _, todaysFortuneCard in
                self.todaysFortuneCard.send(todaysFortuneCard)
            }.store(in: cancelBag)
    }
```

Reply:
넹 감사해요!!